pkg/apps/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -335,30 +308,108 @@ func (c *DeploymentConfigController) reconcileDeployments(existingDeployments []
 // Update the status of the provided deployment config. Additional conditions will override any other condition in the
 // deployment config status.
 func (c *DeploymentConfigController) updateStatus(config *deployapi.DeploymentConfig, deployments []*v1.ReplicationController, additional ...deployapi.DeploymentCondition) error {
-	newStatus := calculateStatus(config, deployments, additional...)
-
-	// NOTE: We should update the status of the deployment config only if we need to, otherwise
-	// we hotloop between updates.
-	if reflect.DeepEqual(newStatus, config.Status) {
-		return nil
+	getFreshCopy := func() (*deployapi.DeploymentConfig, error) {
+		newConfig, err := c.dn.DeploymentConfigs(config.Namespace).Get(config.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		return deployutil.DeploymentConfigDeepCopy(newConfig)
 	}
-
 	copied, err := deployutil.DeploymentConfigDeepCopy(config)
 	if err != nil {
 		return err
 	}
+	var newStatus deployapi.DeploymentConfigStatus
+	// Retry fast on conflicts
+	retryErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		newStatus = calculateStatus(copied, deployments, additional...)
+		// NOTE: We should update the status of the deployment config only if we need to, otherwise
+		// we hotloop between updates.
+		if reflect.DeepEqual(newStatus, copied.Status) {
+			return nil
+		}
 
-	copied.Status = newStatus
-	// TODO: Retry update conficts
-	if _, err := c.dn.DeploymentConfigs(copied.Namespace).UpdateStatus(copied); err != nil {
+		copied.Status = newStatus
+		_, err := c.dn.DeploymentConfigs(copied.Namespace).UpdateStatus(copied)
+		if kapierrors.IsConflict(err) {
+			copied, err = getFreshCopy()
```

Comment:
Say you have just acted on some state (DC), getting 409 you know that state has changed but you have acted on the old version. If you fetch the new config and calculate status from it you will set 
observedGenerationg to the new generation from the fresh copy in calculate status, noe the one you acted on.
Anyone observing that object will think the new state is reconciled because observerGeneration=Generation although just generation-1 has been reconciled.

Reply:
Agreed. Let's not retry DC updates for now.

Reply:
agree, i will remove retrying